providers/os/provider/provider.go

```diff
@@ -381,7 +382,7 @@ func (s *Service) connect(req *plugin.ConnectReq, callback plugin.ProviderCallba
 			conn, err = docker.NewDockerEngineContainer(connId, conf, asset)
 
 		case shared.Type_DockerRegistry.String(), shared.Type_ContainerRegistry.String():
-			conn, err = container.NewRegistryImage(connId, conf, asset)
```

Comment:
this looked odd. we shouldn't be creating an image conn for a registry type

Reply:
Right call